src/Tools/GetDocumentInsider/src/Commands/GetDocumentCommandWorker.cs

```diff
@@ -53,6 +60,88 @@ public int Process()
             return 3;
         }
 
+#if NET7_0_OR_GREATER
+        // Register no-op implementations of IServer and IHostLifetime
+        // to prevent the application server from actually launching after build.
+        void ConfigureHostBuilder(object hostBuilder)
+        {
+            ((IHostBuilder)hostBuilder).ConfigureServices((context, services) =>
+            {
+                services.AddSingleton<IServer, NoopServer>();
+                services.AddSingleton<IHostLifetime, NoopHostLifetime>();
+            });
+        }
+
+        // Register a TCS to be invoked when the entrypoint (aka Program.Main)
+        // has finished running. For minimal APIs, this means that all app.X
+        // calls about the host has been built have been executed.
+        var waitForStartTcs = new TaskCompletionSource<object>();
+        void OnEntryPointExit(Exception exception)
+        {
+            // If the entry point exited, we'll try to complete the wait
+            if (exception != null)
+            {
+                waitForStartTcs.TrySetException(exception);
+            }
+            else
+            {
+                waitForStartTcs.TrySetResult(null);
+            }
+        }
+
+        // Resolve the host factory, ensuring that we don't stop the
+        // application after the host has been built.
+        var factory = HostFactoryResolver.ResolveHostFactory(assembly,
+            stopApplication: false,
+            configureHostBuilder: ConfigureHostBuilder,
+            entrypointCompleted: OnEntryPointExit);
+
+        if (factory == null)
+        {
+            _reporter.WriteError(Resources.FormatMethodsNotFound(
+                HostFactoryResolver.BuildWebHost,
+                HostFactoryResolver.CreateHostBuilder,
+                HostFactoryResolver.CreateWebHostBuilder,
+                entryPointType));
+
+            return 4;
```

Comment:
Start the new `return` integers at `8`. I already used `return 0` through `return 7` in `GetDocumentInsider`.

Reply:
Can you clarify what you mean? This codepath is inside `GetDocumentInsider` and the return codes used here match the ones used in the code that is now in the `else` of this `#ifdef`.

Reply:
The idea was to make the `return` codes completely unique, making it dead easy to find what went wrong.

Reply:
Ah, got it. I've updated these.